### DISCUSSION TITLE
Mach3 Inches Postprocessor

### DISCUSSION THREAD

**Discussion by pagliaso** [fellesverkstedet/Bark-beetle-parametric-toolpaths#88] [General]
Hello,

I have been looking at your wonderful project for some time without much deep, but the other day I studied a little more your help page and was looking at the generated g-code and was very impressed.

I have a very basic CNC machine based on inches and Mach3 and would love to try the generated g-code.

The problem is that my knowledge on post-processors iz ZERO and on programming, nearly ZERO.

I was wondering if you could give me a little orientation of what would need to be done in order to get something that I could run on my machine.

Many thanks for your help and kind regards from Spain,

Alb

**Comment by JensDyvik** [comment]
Hi Alb

Thank you for your kind words. I still use BarkBeetle every week and for about 95% all the milling jobs I do.

I think the default g-code that bark beetle generates should work just fine on Mach3, but I haven't tried. Maybe you can set the Z zero on your machine a little bit above the material and run some test jobs i the air?

Is it on purpose that you have your machine set to work in inches? Maybe it would be nice to switch to millimeters? That should be easy to do in the settings of Mach3.

All the defaults of BarkBeetle are in mm, not inches. But I think as long as you have "G21" as the first line of your g-code from BarkBeetle, it will instruct Mach3 that all of these units are in mm. More info on G21 here: [https://linuxcnc.org/docs/html/gcode/g-code.html](https://linuxcnc.org/docs/html/gcode/g-code.html) I can show you how to make BarkBeetle add this automatically if you need.

Alternatively you can manually set all the bark beetle settings to inch values (like Safe Z height, go to home position, feedrate and plunge rate).

I am off to easter holiday now, but will get back to you next week.

best regards from Jens

**Comment by pagliaso** [comment]
Hi Jens,

Many thanks for your kind reply.

I did as you suggested with a higher Z zero, and as it was looking good, I tried a couple of real jobs, and I am loving what I am getting.

The machine is in inches because it is and American machine that came configured in imperial units, and as I am not a real machinist, it gives me headache to have to think to change that, besides that I read somewhere advising against doing so.

I am modifying all the cutting programs with G21 as you suggests, and everything is working great. It is the fist time that I try this, and I am very happy to be cutting from Rhino and in mm for the first time in my live!

I have a couple more doubts for when you come back ;)

Enjoy your holidays!

Many thanks and best regards,

    Alberto Lagos
